Src/DependencyCollector/Shared/Implementation/HttpCoreDiagnosticSourceListener.cs

```diff
@@ -300,6 +300,10 @@ private void InjectRequestHeaders(HttpRequestMessage request, string instrumenta
                             {
                                 HttpHeadersUtilities.SetRequestContextKeyValue(requestHeaders, RequestResponseHeaders.RequestContextCorrelationSourceKey, sourceApplicationId);
                             }
+                            else
```

Comment:
Talked to @xiaomi7732 in person. One of the examples that makes this concern concrete is when the appId cannot be fetched because the url for resolving the appID is wrong. 

We discussed that we can go either way - that is log the dependency as `HTTP` or `HTTP (tracked dependency)`. The only thing different in either case is the type. In either case the user has to understand what is going on to figure out a fix. While it is possible that, not seeing http tracked dependency will spur them in action to fix the issue. It is also possible that most customers won't even realize that they have a problem because they happily see an http dependency and in fact, seeing http tracked dependency that does not have an appID might make it more obvious to them that they have a problem. 

We feel like it's fine to go either way and decided to keep the change as is in this PR.